datafusion/core/src/datasource/listing/url.rs

```diff
@@ -87,6 +88,34 @@ impl ListingTableUrl {
         }
     }
 
+    /// Perform shell-like path expansions
+    /// * Home directory expansion: "~/test.csv" expands to "/Users/user1/test.csv"
+    /// * Environment variable expansion: "$HOME/$DATA/test.csv" expands to
+    /// "/Users/user1/data/test.csv"
+    fn expand_path_prefix(prefix: &str) -> Result<String, DataFusionError> {
+        let error_msg = format!("Failed to perform shell expansion in path: {prefix}");
+
+        let expanded_dir_output = Command::new("sh")
+            .arg("-c")
+            .arg(&format!("echo {prefix}"))
```

Comment:
This seems like it may be subject to a shell injection attack. 

For example, what if someone did 

```sql
CREATE EXTERNAL TABLE foo LOCATED AT '"\"hi"; rm -rf /"'
```

Could this potentially `rm -rf` the filesystem?

Reply:
Thank you! Totally agreed.
Updated the implementation with "2."